core/commands/add.go

```diff
@@ -185,6 +187,32 @@ You can now check what blocks have been created by:
 
 		events := make(chan interface{}, adderOutChanSize)
 
+		var toadd files.Node = req.Files
+		name := ""
+		if !wrap || pathName != "" {
```

Comment:
IIRC, `pathName` is _just_ for stdin. That is, `echo foobar | ipfs add --stdin-name=foo`. I wonder if this should be handled at the commands-lib layer as well...

Reply:
This actually relates to:

> Another problem I faced but that is not covered by a RFC: when adding content that is not a file (say ipfs add with a data buffer), the body part header Content-Disposition doesn't have a proper filename parameter. While this is legal according to the RFCs, the consequence is that those part are parsed as form values (think text field entered by a user), and not files.

(https://github.com/ipfs/ipfs/issues/395)

We should handle this client-side (technically a breaking change but minor enough that I'm fine with it).